api/types/trustedcluster.go

```diff
@@ -184,6 +184,16 @@ func (c *TrustedClusterV2) SetName(e string) {
 	c.Metadata.Name = e
 }
 
+// Origin returns the origin value of the resource.
+func (c *TrustedClusterV2) Origin() string {
+	return c.Metadata.Origin()
+}
+
+// SetOrigin sets the origin value of the resource.
+func (c *TrustedClusterV2) SetOrigin(origin string) {
+	c.Metadata.SetOrigin(origin)
+}
+
```

Comment:
I believe these are still needed to implement https://github.com/gravitational/teleport/blob/01fc9f0acfde26950c00076180c207aee77f194a/integrations/operator/controllers/reconcilers/legacy_resource_without_labels.go#L29-L37

Reply:
Does the origin mess with the checks we do to only allow updating the role map or the enabled/disabled state?

Reply:
Ahh, yeah. When the operator restarts it will attempt to propagate the origin label and the resource revision. The operator will then fail to reconcile the trusted_cluster with "leaf cluster is already enabled, this update would have no effect"
https://github.com/gravitational/teleport/blob/94f1291569478bf3cea0d5b8ba287657d6814bf8/api/types/trustedcluster.go#L237-L264

Do you know if there is a reason we do not allow an update if does not change either of those fields?

Reply:
I think that we could/should allow updating labels, they don't affect anything else after all. Do IaC tools need a way to only overwrite the resource if its origin is the expected one, or are they fine with just upserting to keep the value up to date?

Reply:
> Do IaC tools need a way to only overwrite the resource if its origin is the expected one

The operator will get the existing resource first and bail out if the resource doesn't have the right origin. Else it will update or upsert based on the resource. The upsert is racy, but this catches most issues.

Tl;dr we need the origin label support, we don't need conditional update based on the origin because we do it IaC side. However users should be allowed to update labels. For example you create a TC, then want the operator to manage it. For the operator to adopt the resource you must edit the resource and add the kubernetes origin.

Reply:
That makes sense. [6243ff2](https://github.com/gravitational/teleport/pull/49789/commits/6243ff2009e9143c33e099934d409e1ea7f881ae) should allow the trusted_cluster labels to be modified now.